controllers/movies.ts

```diff
@@ -0,0 +1,302 @@
+        movies = await query.getMany();
+      }
+      return movies;
+    } catch (err: any) {
+      throw new Error("Movies not found");
+    }
+  }
+
+  async getMovieByImdID(imdb: any) {
+    try {
+      const movie = await Movie.findOne({
+        where: {
+          imdbID: imdb,
+        },
+      });
+      return movie;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async deleteMovie(id: any) {
+    try {
+      const movie = (await Movie.findOne({
+        where: {
+          Id: id,
+        },
+      })) as Movie;
+      await Movie.remove(movie);
+      return movie;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async addMovie(params: MovieParams) {
+    const {
+      Title,
+      Year,
+      Rated,
+      Released,
+      Runtime,
+      Genre,
+      Director,
+      Writer,
+      Actors,
+      Plot,
+      Language,
+      Country,
+      Awards,
+      Poster,
+      Metascore,
+      imdbRating,
+      imdbVotes,
+      imdbID,
+      Type,
+      Response,
+      Images,
+      ComingSoon,
+    } = params;
+
+    try {
+      const movie = await Movie.create({
+        Title: Title,
+        Year: Year,
+        Rated: Rated,
+        Released: Released,
+        Runtime: Runtime,
+        Genre: Genre,
+        Director: Director,
+        Writer: Writer,
+        Actors: Actors,
+        Plot: Plot,
+        Language: Language,
+        Country: Country,
+        Awards: Awards,
+        Poster: Poster,
+        Metascore: Metascore,
+        imdbRating: imdbRating,
+        imdbVotes: imdbVotes,
+        imdbID: imdbID,
+        Type: Type,
+        Response: Response,
+        Images: Images,
+        ComingSoon: ComingSoon,
+      });
+
+      await movie.save();
+      return movie;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async editMovie(params: MovieParams, id: any) {
+    const {
+      Title,
+      Year,
+      Rated,
+      Released,
+      Runtime,
+      Genre,
+      Director,
+      Writer,
+      Actors,
+      Plot,
+      Language,
+      Country,
+      Awards,
+      Poster,
+      Metascore,
+      imdbRating,
+      imdbVotes,
+      imdbID,
+      Type,
+      Response,
+      Images,
+      ComingSoon,
+    } = params;
+
+    try {
+      const movie = (await Movie.findOne({
+        where: {
+          Id: id,
+        },
+      })) as Movie;
+
+      movie.Title = Title;
+      movie.Year = Year;
+      movie.Rated = Rated;
+      movie.Released = Released;
+      movie.Runtime = Runtime;
+      movie.Genre = Genre;
+      movie.Director = Director;
+      movie.Writer = Writer;
+      movie.Actors = Actors;
+      movie.Plot = Plot;
+      movie.Language = Language;
+      movie.Country = Country;
+      movie.Awards = Awards;
+      movie.Poster = Poster;
+      movie.Metascore = Metascore;
+      movie.imdbRating = imdbRating;
+      movie.imdbVotes = imdbVotes;
```

Comment:
Defining the values like here means that we are expecting to receive all of the movie parameters as part of the request. This will not always be the case unless you apply some sort of validation of the request body and define what needs to be sent. Make sure that you are updating only the parameters that are sent, not every parameter of the movie. 
Try testing out how the typeorm save method works, for example if you update only one field, will the other field be removed or not.
Then, try refactoring the code and update only the values of the parameters sent.

Reply:
Fixed it with Partial<MovieParams>